tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -330,4 +330,21 @@ var _ = Describe("odo devfile push command tests", func() {
 		})
 	})
 
+	Context("when .gitignore file exists", func() {
+		It("checks that .odo/env and .odo/odo-file-index.json", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, cmpName)
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(context, "devfile.yaml"))
+
+			output := helper.CmdShouldPass("odo", "push", "--project", namespace)
```

Comment:
Do we need to copy the source and call `odo push` to test this functionality?
Won't this just make the test run longer without any benefit in testing the gitignore functionality?

Reply:
yes we need to do odo push to test this as the ignore file is created as part of `odo push`.

Reply:
so I resolved this by moving the gitignore file creation to the odo create flow and hence the test is simpler now